src/typedefs.ts

```diff
@@ -0,0 +1,11 @@
+interface NominalTag<T> {
+  'nominalTag': T;
+}
+
+type Nominal<Type, Tag> = NominalTag<Tag> & Type;
+
+const enum Degree {}
+const enum Radian {}
+
+export type TDegree = Nominal<number, Degree>;
+export type TRadian = Nominal<number, Radian>;
```

Comment:
This is soething i have seen doing to other devs, is like a away to have 2 numbers that do not mix together.
This is good for radians vs degree and little else probably in our case.

Reply:
I like this

Reply:
If I add an optional mark to the `nominalTag` https://github.com/fabricjs/fabric.js/commit/304767d210aef7c3c9ebcbac051d9258a0394c6a it accepts `number` or `TRadian` but not `TDegree`:
![image](https://user-images.githubusercontent.com/34343793/175014944-25961bf9-f932-4fb4-8fc4-4594afdc70ac.png)

![image](https://user-images.githubusercontent.com/34343793/175014898-7b40af6c-8302-40fd-b81e-e985a9d69b56.png)

Reply:
oh that would be fine, important is that they don't mix when they come out of a function that specifically outputs either radians or degree, i'll make that change.

Typescript should have this partitioned number somehow, even if is not a specific TS use case, but is a nice to have functionality.

Reply:
very nice to have